heka/plugins/s3splitfile/s3splitfile_input.go

```diff
@@ -204,6 +215,17 @@ func (input *S3SplitFileInput) fetcher(runner pipeline.InputRunner, wg *sync.Wai
 	wg.Done()
 }
 
+func (input *S3SplitFileInput) ReportMsg(msg *message.Message) error {
+	message.NewInt64Field(msg, "ProcessFileCount", atomic.LoadInt64(&input.processFileCount), "count")
+	message.NewInt64Field(msg, "ProcessFileFailures", atomic.LoadInt64(&input.processFileFailures), "count")
+	message.NewInt64Field(msg, "ProcessMessageCount", atomic.LoadInt64(&input.processMessageCount), "count")
+	message.NewInt64Field(msg, "ProcessMessageFailures", atomic.LoadInt64(&input.processMessageFailures), "count")
+	message.NewInt64Field(msg, "ProcessMessageBytes", atomic.LoadInt64(&input.processMessageBytes), "B")
+	message.NewInt64Field(msg, "RemainingDataBytes", atomic.LoadInt64(&input.remainingDataBytes), "B")
```

Comment:
From an analysis perspective it is really dropped/discarded/corrupt bytes and it would still fall under your ProcessFile* grouping